### ISSUE TITLE
First child behavior different between usa-grid and usa-grid-full

### ISSUE THREAD

**Issue by jasonkaedingICE** [18F/web-design-standards#1768]
## Description

After upgrading from 0.8.x to 1.0, we observed some differences in the way padding/margin works for grids.  I think, in general, the direction is positive.  However, we have observed that usa-grid and usa-grid-full behave differently.

Quoting the standards website:

> To use a grid without padding on the right and left, use the usa-grid-full class instead.

We found in the CSS, the following sections:
````css
.usa-section > :first-child,
.usa-grid > :first-child {
  margin-top: 0;
}

.usa-section > :last-child,
.usa-grid > :last-child {
  margin-bottom: 0;
}
````

It seems odd that the usa-grid-full selectors weren't added here, given the description of the purpose is to remove left/right padding only.

## Code to demonstrate the difference

````html
<div class="usa-alert usa-alert-info">
  <div class="usa-alert-body">
    <h3 class="usa-alert-heading">Grid</h3>
	<p class="usa-alert-text">There should be no margin between this and the header inside the grid</p>
  </div>
</div>
<div class="usa-grid">
  <h2>In the grid</h2>
  <p></p>
</div>
<div class="usa-alert usa-alert-info">
  <div class="usa-alert-body">
    <h3 class="usa-alert-heading">Full Grid</h3>
    <p class="usa-alert-text">There is extra margin between this and the header inside the grid</p>
  </div>
</div>
<div class="usa-grid-full">
  <h2>In the "full" grid</h2>
  <p></p>
</div>
````

**Comment by shawnbot** [comment]
This is tricky to trace back because the [diff between v0.8.3 and v1.0.0](https://github.com/18F/web-design-standards/compare/v0.8.3...v1.0.0) is huge and the difference in the grid is particularly difficult to disentangle from the rest of the work. However, I _think_ the idea behind the grid is that the `.usa-grid` and `-full` classes aren't intended to host elements other than `.usa-width-*` as direct descendants. In other words, I think you want:

```html
<div class="usa-grid">
  <div class="usa-width-one-whole">
    <h2>In the grid</h2>
    <p>Hey, I'm in the grid!</p>
  </div>
</div>
```

**Comment by jasonkaedingICE** [comment]
So why does the CSS have first-child and last-child for usa-grid at all then?

**Comment by shawnbot** [comment]
Honestly I'm not sure, but my guess is that the [Neat grid](http://neat.bourbon.io/docs/1.8.0/#new-breakpoint) (we're still using v1.8) generates some of those rules automatically.

@juliaelman can probably speak to this better than I can, and has also suggested that we nix Neat (and maybe even Bourbon) altogether, because rolling our own grid would not be that hard.

**Comment by juliaelman** [comment]
@jasonkaedingICE thanks for posting this question. I did some digging as to why this style is in place and it looks like this was left over from when our [docs site](https://github.com/18F/web-design-standards-docs/) repo and this repo were part of the same repo. This is implemented for instances where you would do something like this:

```
<section class="usa-grid usa-section">
  <!--  margin-top: 0 is set to remove the margin value on the <h2>  -->
  <h2>Main head title</h2>
  <div class="usa-width-one-half">
    <h3>Subtitle</h3>
    <p>Your amazing copy about this sub section.</p>
  </div>
  <div class="usa-width-one-half">
    <h3>Subtitle</h3>
    <p>Your amazing copy about this sub section.</p>
  </div>
</section>
```

It makes sense to me to add `.usa-grid-full` to this configuration and am not entirely sure why we excluded.

We have talked about re-making our grid with no dependencies, but it is currently not slated on [our product roadmap](https://standards.usa.gov/whats-new/product-roadmap/). It's something I'd like to see happen (and do!) and appreciate @shawnbot for bringing this up.

And thank you for bringing up this question! The Standards get better and better by users like yourself submitting questions for us to to refine our code base.

Closing this issue and opening a new one to make this change to `typography.scss.`
